src/Advection/weno_reconstruction.jl

```diff
@@ -101,6 +100,11 @@ function WENO(FT::DataType=Oceananigans.defaults.FloatType, FT2::DataType=Float3
 
     mod(order, 2) == 0 && throw(ArgumentError("WENO reconstruction scheme is defined only for odd orders"))
 
+    if !isnothing(bounds)
+        bounds isa Tuple && throw(ArgumentError("bounds must be nothing or a tuple of two values"))
```

Comment:
This reads wrong... It was throwing the ArgumentError if the bounds was actually a Tuple.

How about:

```suggestion
        !(bounds isa Tuple{<:Number, <:Number}) && throw(ArgumentError("bounds must be nothing or a tuple of two values"))
```

Reply:
> This reads wrong... It was throwing the ArgumentError if the bounds was actually a Tuple.

how can we avoid putting the explicit Number?

Reply:
use `NTuple{2}`?